c7n/resources/asg.py

```diff
@@ -598,6 +599,60 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('launch-activity')
+class LaunchActivityFilter(Filter):
+    """Filter ASG based on launch activity status
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: asg-launch-failures-weekly
+                resource: asg
+                filters:
+                  - launch-activity
+                    status: ['Failed']
+                    days: 7
+    """
+    schema = type_schema(
+        'launch-activity',
+        status={
+            'type': 'array',
+            'items': {
+                'enum': [
+                    'Successful', 'Failed', 'Cancelled', 'PreInService',
+                    'InProgress', 'PendingSpotBidPlacement',
+                    'WaitingForSpotInstanceRequestId', 'WaitingForInstanceId',
+                    'WaitingForSpotInstanceId', 'MidLifecycleAction',
+                    'WaitingForELBConnectionDraining',
+                    'WaitingForInstanceWarmup']}},
+        days={'type': 'number'},
+        required=('status',))
+    permissions = ('autoscaling:DescribeScalingActivities',)
+
+    @worker
+    def asg_activities(self, asg):
+        client = local_session(
+            self.manager.session_factory).client('autoscaling')
+        p = client.get_paginator('describe_scaling_activities')
+        for activity in p.paginate(
+                AutoScalingGroupName=asg['AutoScalingGroupName']):
+            for a in activity['Activities']:
+                if self.data.get('days'):
+                    if a['StartTime'].replace(tzinfo=tzutc()) < datetime.now(
+                            tz=tzutc()) - timedelta(days=self.data.get('days')):
+                        break
+                if a['StatusCode'] in self.data.get('status'):
+                    asg.setdefault('c7n:Activities', []).append(a)
+
+    def process(self, asgs, event=None):
+        with self.executor_factory(max_workers=2) as w:
+            for asg_set in utils.chunks(asgs, size=10):
+                list(w.map(self.asg_activities, asg_set))
+        return [a for a in asgs if 'c7n:Activities' in a]
```

Comment:
per previous comments, given two launch event filters and'd this will still cause a single event filter match to always match the other;